array_api_tests/hypothesis_helpers.py

```diff
@@ -129,8 +141,10 @@ def two_broadcastable_shapes(draw, shapes=shapes):
 
 nonbroadcastable_ones_array_two_args = hypotheses_tuples(ones_arrays, ones_arrays)
 
-# TODO: Generate general arrays here, rather than just scalars.
-numeric_arrays = builds(full, just((1,)), floats())
+numeric_arrays = xps.arrays(
```

Comment:
Just a note here: we're going to want to split this across the different [dtype categories](https://github.com/data-apis/array-api-tests/blob/dfe977b9efc0b250bb152f9e23ed591593c3ebf2/array_api_tests/test_type_promotion.py). I guess this strategy is currently used in the special cases tests which are generated automatically, so we would have to also update the generate_stubs script to parse said categories from the spec. I think at present every function that has special cases accepts floating-point inputs which is why this works. This might end up being complicated, so it's something I'll take a look at myself at some point. We are also going to need to consider how things will work with complex dtypes once those start being added later this year.